src/services/specialtyService.ts

```diff
@@ -1,13 +1,50 @@
-import { specialties } from '../mockData/mockData';
-import { Specialty } from '../typeDefs/gqlTypes';
+
+import { Specialty, SpecialtyName } from '../typeDefs/gqlTypes';
+import {Specialty as PrismaSpecialty, SpecialtyName as PrismaSpecialtyName} from '@prisma/client';
+import prisma from '../db/client';
+
+type SpecialtyAndNames = (PrismaSpecialty & { names: PrismaSpecialtyName[]})
+
+function convertPrismaToGraphQLSpecialty(input: SpecialtyAndNames | null) {
+    if (!input) { return null; }
+    const ret = {
+        id: String(input.id),
+        names: Array<SpecialtyName>()
+    } as Specialty;
+
+    for (let i = 0; i < input.names.length; i++) {
+        ret.names?.push({
+            name: input.names[i].name,
+            locale: input.names[i].locale
+        });
+    }
+
+    return ret;
+}
 
 // TODO: add a validation step for incoming parameters
-export const getSpecialtyById = (id: string) => {
-    const matchingSpecialty = specialties.find(
-        (medicalField: Specialty) => medicalField.id === id
-    );
+export const getSpecialtyById = async (id: string) => {
+    const found = await prisma.specialty.findUnique({ where: {
+        id: parseInt(id)
+    },
+    include: {
+        names: true
+    }});
 
-    return matchingSpecialty;
+    return convertPrismaToGraphQLSpecialty(found);
 };
 
-export const getSpecialties = () => specialties;
+export const getSpecialties = async () => {
+    const dbSpecialties = await prisma.specialty.findMany({ include: {
+        names: true
+    }});
+
+    const gqlSpecialties = Array<Specialty>();
+
+    dbSpecialties.forEach(specialty => {
```

Comment:
Here's the example for how you could use `reduce()` here instead. Originally, you could use map to just change each record, but since it's conditionally adding, reduce makes more sense

```suggestion
    const gqlSpecialties  = dbSpecialties.reduce((convertedSpecialties, specialty) => {
        const gqlSpecialty = convertPrismaToGraphQLSpecialty(specialty);
         if (gqlSpecialty) { convertedSpecialties.push(gqlSpecialty); }         
         
         return convertedSpecialties;
}, []);
```

Reply:
^ this is pretty much how most people do array conversions, filtering, etc nowadays. It's effectively the same functionality though, so this is totally up to you which way you prefer

Reply:
I think it's cool and I'd like to study it more, but would feel better when I have better test coverage... right now nothing 😭 

How about I bookmark and return when there are unit or integration tests?